src/ifcparse/IfcParse.cpp

```diff
@@ -2124,15 +2110,8 @@ void IfcFile::process_deletion_() {
 						}
 					}
 					{
-						auto lower = byref_excl.lower_bound({ name,-1,-1 });
-						auto upper = byref_excl.upper_bound({ name, std::numeric_limits<int>::max(), std::numeric_limits<int>::max() });
-
-						for (auto byref_it = lower; byref_it != upper; ++byref_it) {
-							auto& ids = byref_it->second;
-							ids.erase(std::remove(ids.begin(), ids.end(), id), ids.end());
-						}
+						byref_excl.erase(name);
```

Comment:
The issue is here. What's being erased previously wasn't `name`, but rather `id` being erased from the references of `name`. Try this:

						auto byref_it = byref_excl.find(name);
						if (byref_it != byref_excl.end()) {
							auto& ids = byref_it->second;
							ids.erase(std::remove(ids.begin(), ids.end(), id), ids.end());
						}

Reply:
You're right! Thanks and fixed :)